extensions/AbuseFilter/AbuseFilter.hooks.php

```diff
@@ -100,7 +100,9 @@ public static function onAbortMove( $oldTitle, $newTitle, $user, &$error, $reaso
 		return $filter_result == '' || $filter_result === true;
 	}
 
-	public static function onArticleDelete( &$article, &$user, &$reason, &$error ) {
+	public static function onArticleDelete(
+		WikiPage $article, User $user, $reason, &$error
```

Comment:
If `$user` is null `WikiPage::doDeleteArticleReal` will crash anyways when it reaches the call to `LogEntry::setPerformer`, so it does not make much of a difference 😉